File: Data_structures/Array/217. Contains Duplicate2.java
/* Logic: 
Create a set, add elements to it. If an element is already present in
the set return True else return False
Time Complexity - O(N), memory - O(N) (created a set) 
*/

class Solution {
    public boolean containsDuplicate(int[] nums) {
        boolean output = false;
        Set<Integer> numberSet = new HashSet<Integer>();
        for (int i=0; i<nums.length; i++){
            if (numberSet.contains(nums[i])){
                output = true;
                break;
            }
            else{
                numberSet.add(nums[i]);
            }
        }
        return output;
    }
}

File: Data_structures/Array/217. Contains Duplicate2.py
# Logic: 
# Create a set, add elements to it. If an element is already present in
# the set return True else return False
# Time Complexity - O(N), memory - O(N) (created a set)

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        output = False
        number_set = set()
        for element in nums:
            if element in number_set:
                output = True
                break
            else:
                number_set.add(element)    
        return output